imap_processing/swe/l1a_swe.py

```diff
@@ -0,0 +1,183 @@
+            # It should follow by 4 packets where each packet contains
+            # one spin's data.
+            one_full_cycle = np.zeros((24, 30, 7))
+
+        # TODO: get metadata of each data and stores. Find out how to combine four
+        # packet's metadata.
+
+        # read raw data
+        binary_data = decom_data[total_packets].data["SCIENCE_DATA"].raw_value
+        # read raw data as binary array using bitstring
+        bit_array = bitstring.ConstBitStream(bin=binary_data)
+        # chunk binary into 1260 units each with 8-bits
+        byte_data = bit_array.readlist(["uint:8"] * 1260)
+
+        # Uncompress counts. Uncompressed data is a list of 1260
+        # where 1260 = 15 seconds x 7 CEMs x 12 energy steps
+        uncompressed_data = [uncompress(i) for i in byte_data]
+        reshaped_data = np.array(uncompressed_data).reshape(15, 12, 7)
+        # TODO: Confirm how I am populating full data array with four spins data is correct.
+
+        # Reset column index for each spin.
+        column_index = 0
+        # Get indexes for each spin.
+        indexes = get_step_indexes(spin_number)
+
+        for every_second in range(15):
+            # go through first six steps of first half second and populate data in the right index
+            # of one_full_cycle array.
+            for every_step, index_number in enumerate(indexes[0]):
+                one_full_cycle[index_number-1, column_index] = reshaped_data[every_second, every_step]
+            # shift column index by one step.
+            column_index += 1
+            # go through last six steps of second half second and populate data in the right index
+            # of one_full_cycle array.
+            for every_step, index_number in enumerate(indexes[1]):
+                one_full_cycle[index_number-1, column_index] = reshaped_data[every_second, 6 + every_step]
+            # shift column index by one step.
+            column_index += 1
+        # If spin number is 3, then we have got all four spins data.
+        if spin_number == 3:
+            # Reshape data to (24, 7, 30)
+            one_full_cycle = np.transpose(one_full_cycle, (0, 2, 1))
+            # TODO: write data to CDF file
+            metadata = {}
+            for key, value in decom_data[total_packets].header.items():
+                metadata[key] = value.raw_value
+
+            for key, value in decom_data[total_packets].data.items():
+                if key != "SCIENCE_DATA":
+                    metadata[key] = (
+                        value.raw_value
+                        if value.derived_value is None
+                        else value.derived_value
+                    )
+            print(one_full_cycle[:4, 0, :])
+            xarray_data = xr.DataArray(
+                list(one_full_cycle), dims=["energy_step", "cem_counts", "angle"],
+            )
+            dataset = xr.Dataset({"data": xarray_data})
+            dataset.to_netcdf(f"swe_l1a_{total_packets}.nc")
```

Comment:
:( I wish we could use netcdf too! I don't think we should keep this as a dependency, unfortunately. One thought would be to add it to testing dependencies.